SU-HSDO/suhumsci
# feat: added breakout box config and styles

## Summary
This pull request gives content editors the ability to add "Breakout Box" paragraph styles to pages. 

This pull request adds the following:
- Config for CKEditor/WYSIWYG types.
- Styling for Humsci Basic themes.
- New class: (`hs-well`) and styling for Humsci Legacy theme.

I opted to use the class ('hs-well') instead of ('hb-well') to keep this consistent with other inline paragraph syles within WYSIWYG/CKEditor. They use the same code for styling, but it was updated to a mixin since it has a more universal use. 
Info detailed in [this Jira card.](https://sparkbox.atlassian.net/browse/STN-365)

## Browsers Tested
The following browser testing has been completed:

## Steps to Test
1. Run npm test
2. Import changes to the ckeditor config files.
3. Create a basic page, (content > basic page) and add Breakout Box text to the page within the WYSIWYG editor style options.
4. Confirm the styling for Breakout Box https://sparkbox-sandbox-prod.stanford.edu/qa/breakout-and-quote-styles shows up in the WYSIWYG while content editing and the option in text styles exists.
![image](https://user-images.githubusercontent.com/9355698/88112431-dc900d00-cb7d-11ea-9041-3008f10791e1.png)
5. Confirm the styling shows up correctly on the page view after saving the page.
![image](https://user-images.githubusercontent.com/9355698/88112377-bff3d500-cb7d-11ea-9049-60fcdef838cc.png)
6. Add a variety of style options as outlined in the Jira card, within a row, within an accordion, etc.
7. Test that same page in the different appearance themes: Colorful, Traditional and the Legacy theme.

(I was able to test images floating next to breakout boxes in rows, but they did not allow placement within the breakout box. If this functionality needs changed, I will want to pair with someone to discuss this ability.)

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
